test/hotspot/gtest/nmt/test_vmatree.cpp

```diff
@@ -168,6 +168,36 @@ class NMTVMATreeTest : public testing::Test {
     EXPECT_TRUE(exists(found[2]));
     EXPECT_TRUE(exists(found[3]));
   };
+  void copy_flag_test() {
+    {
+      Tree tree;
+      VMATree::RegionData rd1{ si[0], mtTest };
+      VMATree::RegionData rd2{ si[1], mtNMT };
+
+      tree.reserve_mapping(0, 100, rd1);
+      tree.commit_mapping(20, 50, rd2, true); // mtTest flag is to be copied to new nodes, even if it is given as mtNMT.
```

Comment:
When does this happen, that we reserve an overlapping region with a different NMT flag?

Reply:
Here in this test, I used a different flag to verify that the new flag is not used for the new nodes of the tree. This is testing even if a flag is given but the arg is true, it will be ignored.

Reply:
> When does this happen, that we reserve an overlapping region with a different NMT flag?

Uncommitting is essentially the same as reserving. We don't do this in the codebase right now, because that'd make NMT crash, but there's nothing weird about the pattern itself.

Reply:
Reserving over already reserved memory would replace the old mapping and thus destroy the old content (as does committing, btw). So, this is rather rare. When its not an error, it is done to e.g. change mapping properties before starting to use the mapping.

Reply:
OK thank you, just wanted to make sure that this usage is what we want and allow. I initially thought we would not want that and have some assert checking for it.